junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/descriptor/TestMethodTestDescriptor.java

```diff
@@ -231,4 +234,53 @@ private void invokeAfterEachCallbacks(JupiterEngineExecutionContext context) {
 		});
 	}
 
+	@Override
+	public void nodeSkipped(JupiterEngineExecutionContext context, TestDescriptor descriptor, SkipResult result) {
+		if (context != null) {
```

Comment:
When would `context` be `null` here and below?

Reply:
Context was null in the tests launched by the ExceptionHandlingTests and that caused them to fail with a NullPointerException on context. I have not looked into the exact cause yet(will do), but there is another check for a null context in NodeTestTask.

Reply:
I checked and `context` is indeed `null` when `prepare()` throws an exception which happens when an exception occurs creating the test instance.

If we want to report such failed tests to extensions as well and I think we should, we need to change `prepare` to add the exception to `ThrowableCollector` instead and add `context.getThrowableCollector().assertEmpty()` to `JupiterTestDescriptor.shouldBeSkipped()` (otherwise a test with a failing constructor could be reported as skipped).

One more thing I noticed: When something fails on the test class level, e.g. an exception a `@BeforeAll` method, `TestMethodTestDescriptor` never gets called. I think we should clearly document this limitation in the Javadoc of `TestWatcher`.

Reply:
Sounds reasonable to me, @marcphilipp

Reply:
And... thanks for investigating!